build/src/lib.rs

```diff
@@ -420,42 +600,44 @@ mod tests {
     }
 
     #[test]
-    fn binary_builder_works() -> Result<(), Error> {
+    fn binary_builder_works() -> Result<()> {
         let tempdir = tempdir::TempDir::new("builder_test")?;
         let tempdir = tempdir.path();
 
-        BinaryBuilder::new(None, tempdir)
+        BinaryBuilder::new(&[], tempdir)
             .fallback_to_build_dir(true)
-            .build()
-            .validate();
+            .build()?
+            .validate()?;
 
         Ok(())
     }
 
     #[test]
-    fn binary_builder_works_with_released_version() -> Result<(), Error> {
+    fn binary_builder_works_with_released_version() -> Result<()> {
         let tempdir = tempdir::TempDir::new("builder_test")?;
         let tempdir = tempdir.path();
 
-        BinaryBuilder::new(Some(&["en"]), tempdir)
+        BinaryBuilder::new(&["en"], tempdir)
             .version("0.3.0")
-            .build();
+            .build()?;
 
         Ok(())
     }
 
     #[test]
-    fn binary_builder_works_with_smush() -> Result<(), Error> {
+    fn binary_builder_works_with_smush() -> Result<()> {
         let tempdir = tempdir::TempDir::new("builder_test")?;
         let tempdir = tempdir.path();
 
-        BinaryBuilder::new(Some(&["en"]), tempdir)
+        BinaryBuilder::new(&["en"], tempdir)
             .version("0.3.0")
-            .build()
+            .build()?
             .postprocess(
-                |buffer, mut writer| {
+                |mut buffer, mut writer| {
+                    let mut tmp = Vec::new();
+                    buffer.read_to_end(&mut tmp)?;
                     Ok(writer.write_all(&smush::encode(
-                        &buffer,
+                        &mut tmp,
```

Comment:
`&mut` is not necessary here.

```suggestion
                        &tmp,
```